Add checkAjax support in Controller::redirect

ivanovyordan opened the pull request:
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | yes |
| Tests pass? | yes |
| Fixed issues | #11058 |

ivanovyordan committed:
Add checkAjax support in Controller::redirect

Resolves #11058

SilverFire commented on framework/web/Controller.php:
```diff
@@ -139,11 +139,17 @@ public function beforeAction($action)
+     *
+     * @param boolean $checkAjax whether to specially handle AJAX (and PJAX) requests. Defaults to true,
+     * meaning if the current request is an AJAX or PJAX request, then calling this method will cause the browser
+     * to redirect to the given URL. If this is false, a `Location` header will be sent, which when received as
```
`false` -> <code>\`false\`</code>

ivanovyordan replied:
Setting default value to `false` would break BC right?

SilverFire replied:
Yeah

ivanovyordan replied:
@SilverFire I made the changes.

Thanks for the feedback guys.

SilverFire reviewed (changes requested):
CHANGELOG and UPGRADE.md should be updated

ivanovyordan committed:
Set `checkAjax` to false

Fix as mentioned in https://github.com/yiisoft/yii2/pull/12650#discussion_r81477396

ivanovyordan committed:
Update CHANGELOG

rob006 commented:
This breaks BC anyway, since it changes method signature.

samdark commented:
Agree. It's a BC break.

SilverFire commented:
I'm fine with merging into 2.1 right now

samdark commented:
Merged into 2.1: https://github.com/yiisoft/yii2/commit/0d72e37f3ce759a7286423eb32f19814bc2e0392
